presto-main/src/main/java/com/facebook/presto/metadata/SignatureBinder.java

```diff
@@ -646,21 +634,7 @@ public SolverReturnStatus update(BoundVariables.Builder bindings)
 
         private boolean satisfiesConstraints(Type type)
         {
-            if (comparableRequired && !type.isComparable()) {
-                return false;
-            }
-            if (orderableRequired && !type.isOrderable()) {
-                return false;
-            }
-            if (!typeBound.isInstance(type)) {
-                return false;
-            }
-            if (requiredBaseName.isPresent() && !UNKNOWN.equals(type) && !requiredBaseName.get().equals(type.getTypeSignature().getBase())) {
-                // TODO: the case below should be properly handled:
-                // * `type` does not have the `requiredBaseName` but can be coerced to some type that has the `requiredBaseName`.
-                return false;
-            }
-            return true;
+            return typeVariableConstraint.canBind(type);
```

Comment:
Should just remove this private function and directly call `typeVariableConstraint.canBind`

Reply:
The two function handles UNKNOWN differently. Is this change correct?

Reply:
The code looks different but the logic is same (I think).

Reply:
Can you run a verifier explain test to double check on this?

Reply:
Can we delete this function now?

Reply:
I thought I did !

Reply:
Done for real now :)